gslib/utils/boto_util.py

```diff
@@ -499,6 +485,49 @@ def ResumableThreshold():
   return config.getint('GSUtil', 'resumable_threshold', 8 * ONE_MIB)
 
 
+def SetProxyInfo():
+  """Sets proxy info from boto and environment and converts to httplib2.ProxyInfo.
+
+  Args:
+    None.
+
+  Returns:
+    httplib2.ProxyInfo constructed from boto or environment variable string.
+  """
+  #Defining proxy_type based on httplib2 library, accounting for None entry too.
+  proxy_type_spec = {'socks4': 1, 'socks5': 2, 'http': 3, 'https': 3}
+
+  #proxy_type defaults to 'http (3)' for backwards compatibility
+  proxy_type = proxy_type_spec.get(config.get('Boto', 'proxy_type', None)) or 3
+  proxy_host = config.get('Boto', 'proxy', None)
+
+  #For proxy_info below, proxy_rdns fails for socks4 and socks5 so restricting use
+  #to http only
+  proxy_info = httplib2.ProxyInfo(
+      proxy_host=proxy_host,
+      proxy_type=proxy_type,
+      proxy_port=config.getint('Boto', 'proxy_port', 0),
+      proxy_user=config.get('Boto', 'proxy_user', None),
+      proxy_pass=config.get('Boto', 'proxy_pass', None),
+      proxy_rdns=config.getbool('Boto', 'proxy_rdns',
+                                True if proxy_type == 3 else False))
```

Comment:
use `proxy_type_spec.https` instead of `3`

Reply:
Changed line to: 
`True if proxy_type == proxy_type_spec['http'] else False))`